api/server/sdk/cluster_pair.go

```diff
@@ -0,0 +1,135 @@
+/*
+Package sdk is the gRPC implementation of the SDK gRPC server
+Copyright 2018 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package sdk
+
+import (
+	"context"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/cluster"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ClusterPairServer is an implementation of the gRPC OpenStorageClusterServer interface
+type ClusterPairServer struct {
+	server *Server
+}
+
+func (s *ClusterPairServer) cluster() cluster.Cluster {
+	return s.server.cluster()
+}
+
+// Create a new cluster with remote pair
+func (s *ClusterPairServer) Create(
+	ctx context.Context,
+	req *api.SdkClusterPairCreateRequest,
+) (*api.SdkClusterPairCreateResponse, error) {
+
+	if req.GetRequest() == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Must supply valid request")
+	}
+
+	resp, err := s.cluster().CreatePair(req.GetRequest())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot create cluster with remote pair %s : %v",
+			req.GetRequest().GetRemoteClusterIp(), err)
+	}
+
+	return &api.SdkClusterPairCreateResponse{
+		Result: resp,
+	}, nil
+}
+
+// Inspect information about a cluster pair
+func (s *ClusterPairServer) Inspect(
+	ctx context.Context,
+	req *api.SdkClusterPairInspectRequest,
+) (*api.SdkClusterPairInspectResponse, error) {
+
+	if len(req.GetId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply cluster ID")
+	}
+	resp, err := s.cluster().GetPair(req.GetId())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot Get cluster information for %s : %v", req.GetId(), err)
+	}
+	return &api.SdkClusterPairInspectResponse{
+		Result: resp,
+	}, nil
+}
+
+// Enumerate returns list of cluster pairs
+func (s *ClusterPairServer) Enumerate(
+	ctx context.Context,
+	req *api.SdkClusterPairEnumerateRequest,
+) (*api.SdkClusterPairEnumerateResponse, error) {
+
+	resp, err := s.cluster().EnumeratePairs()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot list cluster pairs : %v", err)
+	}
+	return &api.SdkClusterPairEnumerateResponse{
+		Result: resp,
+	}, nil
+}
+
+// Token gets the authentication token for this cluster
+func (s *ClusterPairServer) Token(
```

Comment:
Just Token() doesn't really make sense. This is like saying Volume(). It should be GetToken()

Reply:
I think this is clear and makes sense. It is like saying time.Now().Unix(), returns time in Unix() format. This would say `migrate.Token()` which returns a token. Seems perfectly aligned.

Reply:
We don't do Volume() or Cluster() right? So why have different conventions for different APIs? It should be consistent I think. There is no other API here that uses just a noun for the get call.

Reply:
There is an Volume.Inspect(), is there not? Having just a noun as the API doesn't make sense. Using old Unix APIs is not a valid argument.

Reply:
i thought more on the Token vs GetToken, and i think GetToken() is fine. That is what the golang proto buf does for every value anyways.. GetXXX().